app/components/SearchableMetricSelector.js

```diff
@@ -58,11 +92,109 @@ function Modal(props){
     )
 }
 
+function SimpleMetricSearchResults(props){
+  const { searchText, supportedMetrics, onClose, selectMetric } = props
+  const searchTextLower = searchText.toLowerCase().trim()
+  const searchResults = searchTextLower.length === 0 
+    ? Object.keys(supportedMetrics)
+    : Object.keys(supportedMetrics)
+      .filter((k) => {
+        const metric = supportedMetrics[k]
+        const metricNameLower = metric.name.toLowerCase()
+        return metricNameLower.indexOf(searchTextLower) > -1
+      })
+      
+  return ( 
+    <div className="searchResults">
+      {searchResults.map((k, i) => (
+        <p
+          className="metricResult"
+          key={i} 
+          onClick={ () => {
+            selectMetric(k)
+            onClose()  
+          }}
+        >
+          {supportedMetrics[k].name}
+            <ul className="metricMetaData">
+              <li>Dataset: {supportedMetrics[k].dataset}</li>
+              <li>Description: {supportedMetrics[k].description}</li>
+            </ul>
+          </p>
+              ))}
+    </div>
+  )
+  
+}
+
+function GroupedMetricSearchResults(props){
+  const { searchText, supportedMetrics, onClose, selectMetric, tagDictionary, tagDictArray } = props
+  const searchTextLower = searchText.toLowerCase().trim()
+  const searchResults = searchTextLower.length === 0 
+    ? Object.keys(supportedMetrics)
+    : Object.keys(supportedMetrics)
+      .filter((k) => {
+        const metric = supportedMetrics[k]
+        const metricNameLower = metric.name.toLowerCase()
+        return metricNameLower.indexOf(searchTextLower) > -1
+      })
+
+  var elements = [];
+  for (var i = 0; i < tagDictArray.length; i++){
+    for (var key in tagDictArray[i]){
+      const metricArr = tagDictionary[key]
+      var count = 0;
+      for (var j = 0; j < searchResults.length; j++){
+        if (supportedMetrics[searchResults[j]].dataset.indexOf(key) > -1 && count < 1){
```

Comment:
Why do you have `count < 1`?

Reply:
This `count < 1` dictates that each group is only printed once. I'm sure there is a better way of doing this, previously it would print the rideshare group once for every rideshare metric in the dictionary

Reply:
Ah I see. Thanks for the explanation. Can you add a comment to explain that? We can refactor it in the future.